navigator/runtime/src/main/java/com/freeletics/mad/navigator/DeepLink.kt

```diff
@@ -0,0 +1,136 @@
+     *
+     * If an [action] is provided the [Intent] returned by [buildIntent] and the other builders
+     * will use it as it's [Intent.getAction]. If not `action` is provided the app's
+     * [android.content.pm.PackageManager.getLaunchIntentForPackage] will be used.
+     */
+    public constructor(
+        root: NavRoot,
+        routes: List<NavRoute> = emptyList(),
+        action: String? = null,
+    ) : this(action, listOf<BaseRoute>(root) + routes)
+
+    /**
+     * A deep link created with this will open the app with it's start destination and then launch
+     * [ActivityRoute] on top of it.
+     *
+     * If an [action] is provided the [Intent] returned by [buildIntent] and the other builders
+     * will use it as it's [Intent.getAction]. If not `action` is provided the app's
+     * [android.content.pm.PackageManager.getLaunchIntentForPackage] will be used.
+     */
+    public constructor(
+        activityRoute: ActivityRoute,
+        action: String? = null
+    ) : this(action, listOf(activityRoute))
+
+    /**
+     * A deep link created with this will open the app with the given [routes] added to the back
+     * stack on top of the start destination, [activityRoute] will then be launched on top of this.
+     *
+     * If an [action] is provided the [Intent] returned by [buildIntent] and the other builders
+     * will use it as it's [Intent.getAction]. If not `action` is provided the app's
+     * [android.content.pm.PackageManager.getLaunchIntentForPackage] will be used.
+     */
+    public constructor(
+        routes: List<NavRoute>,
+        activityRoute: ActivityRoute,
+        action: String? = null
+    ) : this(action, routes + activityRoute)
+
+    /**
+     * A deep link created with this will open the app and create a back stack with [root] on top
+     * of the start destination. The given [routes] will be added to that back stack and
+     * [activityRoute] will then be launched on top of this.
+     *
+     * If an [action] is provided the [Intent] returned by [buildIntent] and the other builders
+     * will use it as it's [Intent.getAction]. If not `action` is provided the app's
+     * [android.content.pm.PackageManager.getLaunchIntentForPackage] will be used.
+     */
+    public constructor(
+        root: NavRoot,
+        routes: List<NavRoute>,
+        activityRoute: ActivityRoute,
+        action: String? = null,
+    ) : this(action, listOf<BaseRoute>(root) + routes + activityRoute)
+
+    /**
+     * Creates an [Intent] that that can be used to launch this deep link.
+     */
+    public fun buildIntent(context: Context): Intent {
+        val intent = if (action != null) {
+            Intent(action).setPackage(context.packageName)
+        } else {
+            requireNotNull(context.packageManager.getLaunchIntentForPackage(context.packageName)) {
+                "Couldn't obtain launch intent for ${context.packageName}"
+            }
```

Comment:
This is new and allows single activity apps to get away without defining any custom action to use.